controller/src/test/java/io/pravega/controller/task/Stream/StreamMetadataTasksTest.java

```diff
@@ -458,6 +461,294 @@ public void retentionStreamTest() throws Exception {
+
+        List<StreamCutRecord> list = streamStorePartialMock.getStreamCutsFromRetentionSet(SCOPE, streamName, null, executor).get();
+        assertTrue(list.contains(streamCut1));
+        // endregion
+
+        // region stream cut exists but latest - previous < retention.size
+        // second retention iteration
+        // streamcut2: 100 bytes(0/50, 1/50)
+        Map<Integer, Long> map2 = new HashMap<>();
+        map2.put(0, 50L);
+        map2.put(1, 50L);
+        long recordingTime2 = recordingTime1 + Duration.ofMinutes(Config.MINIMUM_RETENTION_FREQUENCY_IN_MINUTES).toMillis() + 1;
+        size = streamStorePartialMock.getSizeTillStreamCut(SCOPE, streamName, map2, null, executor).join();
+        assertEquals(size, 100L);
+        StreamCutRecord streamCut2 = new StreamCutRecord(recordingTime2, size, map2);
+        doReturn(CompletableFuture.completedFuture(streamCut2)).when(streamMetadataTasks).generateStreamCut(
+                anyString(), anyString(), any());
+
+        streamMetadataTasks.retention(SCOPE, streamName, retentionPolicy, recordingTime2, null).get();
+        list = streamStorePartialMock.getStreamCutsFromRetentionSet(SCOPE, streamName, null, executor).get();
+        StreamProperty<StreamTruncationRecord> truncProp = streamStorePartialMock.getTruncationProperty(SCOPE, streamName, true, null, executor).get();
+        // verify that only one stream cut is in retention set. streamCut2 is not added
+        // verify that truncation did not happen
+        assertTrue(list.contains(streamCut1));
+        assertTrue(list.contains(streamCut2));
+        assertTrue(!truncProp.isUpdating());
+        // endregion
+
+        // region latest - previous > retention.size
+        // third retention iteration
+        // streamcut3: 120 bytes(0/60, 1/60)
+        Map<Integer, Long> map3 = new HashMap<>();
+        map3.put(0, 60L);
+        map3.put(1, 60L);
+        size = streamStorePartialMock.getSizeTillStreamCut(SCOPE, streamName, map3, null, executor).join();
+        assertEquals(size, 120L);
+
+        long recordingTime3 = recordingTime2 + Duration.ofMinutes(Config.MINIMUM_RETENTION_FREQUENCY_IN_MINUTES).toMillis() + 1;
+        StreamCutRecord streamCut3 = new StreamCutRecord(recordingTime3, size, map3);
+        doReturn(CompletableFuture.completedFuture(streamCut3)).when(streamMetadataTasks).generateStreamCut(
+                anyString(), anyString(), any());
+
+        streamMetadataTasks.retention(SCOPE, streamName, retentionPolicy, recordingTime3, null).get();
+        // verify two stream cuts are in retention set. Cut 1 and 3.
+        // verify that Truncation has happened.
+        list = streamStorePartialMock.getStreamCutsFromRetentionSet(SCOPE, streamName, null, executor).get();
+        truncProp = streamStorePartialMock.getTruncationProperty(SCOPE, streamName, true, null, executor).get();
+
+        assertTrue(!list.contains(streamCut1));
```

Comment:
The comment of line 541 implies that Cut 1 should actually be in. This assertion seems to be inconsistent with respect to the comment.

Reply:
copy paste..  its the comment that is wrong. fixed it